Tests/SwiftStdlibTests/OptionalExtensionsTests.swift

```diff
@@ -63,6 +63,24 @@ final class OptionalExtensionsTests: XCTestCase {
 		XCTAssert(parameters[key2] == parameter2)
 	}
 
+    func testConditionalAssignment() {
```

Comment:
What happens if you do multiple `?=` calls in 1 line, e.g.
```Swift
a ?= b ?= c
```
when `a` and `b` are `nil` but `c` has a value. Will `a` get the value of `c`? Will `b` also get its value? Can you test for that?
And if `b` is not `nil`, will `a` become `b` and `b` keep its value? Please test for that too.
And what if `a` is not `nil`, `b` is `nil` and `c` is not `nil`, will `b` become `c`?

Reply:
It won't compile, because `?=` operator just like the assignment operator doesn't return any value ('a = b = c' won't compile too).

Reply:
```
a ?= (b ?= c)
``` 
should work  as @guykogus  explained. 

Could you add the tests for those scenarios?

Reply:
My original idea was to create a conditional assignment operator with the same properties as the original assignment operator (and usual assignment operator doesn't return any value, so `a = (b = c)` won't compile.
On the other hand, the idea of the operator is borrowed from Ruby's conditional assignment operator `||=` and in Ruby assignment operators do return values.

I think that we should stick with Swift's paradigm of non-returning assignments for the sake of consistency.